pkg/trie/triedb/proof/verify.go

```diff
@@ -0,0 +1,406 @@
+// Copyright 2024 ChainSafe Systems (ON)
+// SPDX-License-Identifier: LGPL-3.0-only
+
+package proof
```

Comment:
We should probably have tests for this file right?

Reply:
We are testing both generation and proofs in `proof_test.go`